src/async-snowplow.ts

```diff
@@ -0,0 +1,119 @@
+export type SnowplowFn = (...args: any[]) => void;
+
+/**
+ * Wraps Snowplow method and allows for flushing a bunch of Snowplow calls.
+ */
+export interface AsyncSnowplow {
+  /**
+   * Same as calling Snowplow but will queue the events if Snowplow is not loaded.
+   */
+  call(...args: any[]): void;
```

Comment:
Would a more explicit name help, like queueEvent or logEvent?

Reply:
I'd be okay with `callSnowplow`.  The Snowplow method is a little crazy.  It has some weird extra calling patterns it supports.  E.g. it can be passed a function and then Snowplow creates a context around it and passes in args.  Thoughts?

Reply:
"callSnowplow" sounds good. "call" was a bit generic.

Reply:
Ok.  Done.